waku/v2/node/waku_node.nim

```diff
@@ -544,8 +543,8 @@ proc executeMessageRetentionPolicy*(node: WakuNode) =
 
   debug "executing message retention policy"
 
-  node.wakuArchive.executeMessageRetentionPolicy()
-  node.wakuArchive.reportStoredMessagesMetric()
+  asyncSpawn node.wakuArchive.executeMessageRetentionPolicy()
+  asyncSpawn node.wakuArchive.reportStoredMessagesMetric()
```

Comment:
Not sure of impact, but since this is a timed task, do we want to spawn more futures here or shouldn't this function block until we have finished executing the retention policy as it did before (i.e. use `waitFor` rather than `asyncSpawn`)?